File: Assets/Scripts/Shooter.cs
using UnityEngine;
using System.Collections;

public class Shooter : MonoBehaviour {

	public Rigidbody bullet; //what makes the projectile
	public float power = 1500f; //floats are decimals
	public float moveSpeed = 2f; //and must end with f

	// Update is called once per frame
	void Update () {

		float h = Input.GetAxis("Horizontal") * Time.deltaTime * moveSpeed; //will control the x axis [<-- --> or a d] for movement
		float v = Input.GetAxis("Vertical") * Time.deltaTime * moveSpeed; //will control the y axis [^ v or w s] for movement

		transform.Translate(h,v,0);

		if(Input.GetButtonUp("Fire1")){ //if left click is pressed, fire a projectile

			Rigidbody instance = Instantiate(bullet, transform.position, transform.rotation) as Rigidbody; //will create the projectile
			Vector3 fwd = transform.TransformDirection(Vector3.forward); //this will make the bullet shoot forward

			instance.AddForce(fwd * power); //will give the bullet the ability to go forward
		}
	}
}

File: Assets/Scripts/Shooter.js
#pragma strict

var bullet : Rigidbody; //creates the projectile
var power : float = 1500; //this is the stregnth of the projectile
var moveSpeed : float = 5; //this is how fast the projectile moves

function Update () {

	var h : float = Input.GetAxis("Horizontal") * Time.deltaTime * moveSpeed; //will control the x axis [<-- --> or a d ]for movement
	var v : float = Input.GetAxis("Vertical") * Time.deltaTime * moveSpeed; //will control the y axis [^ v or w s] for movement

	transform.Translate(h,v,0);

	if(Input.GetButtonUp("Fire1")){ //if left click is pressed, fire

		var instance: Rigidbody = Instantiate(bullet, transform.position, transform.rotation); //will create the projectile
		var fwd: Vector3 = transform.TransformDirection(Vector3.forward); //this will make the bullet shoot forward
	
		instance.AddForce(fwd * power); //will give the bullet the ability to go forward
	}	

}